### DISCUSSION TITLE
Not correct converted anchor link

### DISCUSSION THREAD

**Discussion by h7b** [ObsidianPublisher/obsidian-github-publisher#131] [Q&A — Template] [answered]
I doubt whether this problem relates to mkdocs-material template or the plugin itself. 

My question relates to the conversion of a wikilink from Obsidian into an anchor link in published page.

I want to know whether there is an option to render all internal links (anchor link to heading, or note) in a slugified version? (that is all character are lower case, `space` is replaced by hyphen `-`)

## What I did

I have this content in a `note.md` in Obsidian vault
```md
I removed the [[image#Resize image and display|css tag to resize image to 300px supported by Dendron]], since Obsidian use a different syntax
```
When publishing with plugin, the wikilink in the `note.md` is converted into a markdown link as showed in the next line. (Please find [here](https://github.com/h7b/h7b-public-mkdocs/blob/main/docs/pages/dendron/migrate-dendron-obsidian.md?plain=1#L23), the uploaded-to-repo note that I'm discussing)
```md
I removed the [css tag to resize image to 300px supported by Dendron](./image.md#Resize%20image%20and%20display), since Obsidian use a different syntax
```
Then the anchor link is rendered as follows. (that is: `%20` for `space`, character keep its format - upper case lower case - as in Obsidian)
```
https://h7b1.netlify.app/pages/dendron/image/#Resize%20image%20and%20display
```
Which is not the correct anchor link, as pointed below. (that is: `space` is replaced by `-`, and all characters are lower case)
```
https://h7b1.netlify.app/pages/dendron/image/#resize-image-and-display
```

**Comment by Mara-Li** [comment]
Hello!
Maybe it's what you want : https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#+toc.slugify

Also, I want to know if you transform wiki to markdown links?

I already do some test around with markdown links and space without noticed error... So I don't understand the problem 😭

**Reply by h7b** [reply]
> Maybe it's what you want : https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#+toc.slugify

yes. I already applied this into `mkdocs.yml`

> Also, I want to know if you transform wiki to markdown links?

you mean the option `[[Wikilinks]] to [MDlinks](links)` in the settings of `GitHub Publisher` plugin? Then yes, i enabled it.

## Fixed

With your suggestion, I found that the option `[[Wikilinks]] to [MDlinks](links)` is the problem.

When it is disabled, the upload-to-repo note keeps the wikilink format instead of converting into mdlink.
```md
[[./image#Resize image and display|css tag to resize image to 300px supported by Dendron]]
```
Then the `toc.slugify` in `mkdocs.yml` have rendered correctly the slugified anchor link as expected.
```
https://h7b1.netlify.app/pages/dendron/image/#resize-image-and-display
```

**Reply by Mara-Li** [reply]
Normally, the mkdocs use ENCODING instead of SLUGIFY for rendering links.
I don't have any issue with image in the default parameters with encoding. Instead, slugify are ignored.

Why do you want to use slug instead of the default encoding?

**Reply by h7b** [reply]
for example,

the toc link from your docs
```
https://obsidian-publisher.netlify.app/github%20publisher/local%20folder/
```
If I enable the `toc.slugify`, it should be
```
https://obsidian-publisher.netlify.app/github-publisher/local-folder/
```
> I don't have any issue with image in the default parameters with encoding. Instead, slugify are ignored.

There's nothing related to image rendering here. It's just the name of my example link. It's the anchor link to heading in a note that I'm talking about  :D

**Reply by Mara-Li** [reply]
I really don't understand the problem...
I mean, only anchor (so link to an heading/in the toc) will be slug so what is the relation with wiki/markdown links ?

**Reply by h7b** [reply]
> I mean, only anchor (so link to an heading/in the toc) will be slug so what is the relation with wiki/markdown links ?

I don't know either.

I just stated my experience. Anyway, the problem is already fixed. So you should not be wasted your time here :D

If you are curious, allow me to paraphrase and reproduce the experiment.

With a raw note in obsidian. There's a wikilink pointing to the heading `Resize image and display` in the note `image.md`
```md
I removed the [[./image#Resize image and display|css tag to resize image to 300px supported by Dendron]], since Obsidian use a different syntax
```
When the option `[[Wikilinks]] to [MDlinks](links)` is enabled, this content is converted to
```md
I removed the [css tag to resize image to 300px supported by Dendron](./image.md#Resize%20image%20and%20display), since Obsidian use a different syntax
```
where the anchor link in the published page is
```md
I removed the [css tag to resize image to 300px supported by Dendron](https://h7b1.netlify.app/pages/dendron/image/#Resize%20image%20and%20display
), since Obsidian use a different syntax
```
which is not a correct anchor link. It just point to the file `image.md`, but not the correct heading.

When the option `[[Wikilinks]] to [MDlinks](links)` is disabled, the slugified anchor link is generated as my expectation.

In my amateur blind guess, I think that is the slugify option applies correctly only to wikilinks, but not mdlinks. That's how I'm seeing the relation between slugified anchor link in a published page and wikilink/mdlink in the source note.

**Reply by Mara-Li** [reply]
Oh yeah, I understand now!
Maybe, without touching the plugin (in obsidian) adding a different slug for the toc can prevent breaking.

If I can't change that, I will update the plugin to add an option to use slug wikilinks instead of "encoding" wikilinks, for title / anchor.

Thanks!

**Comment by Mara-Li** [comment, marked as answer]
Hello!
I created a new option"slugify anchor for markdown links", could you try it ? 
Thanks!

**Reply by h7b** [reply]
Great.

So with this new option, I don't need to enable the [toc.slugify](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#+toc.slugify) extension in `mkdocs.yml`. Just enable only this new `slugify` option in the `GitHub Publisher` plugin. It's what I tried.

And this `slugify` option work best with enabling the option `wikilink to mdlink`. Somehow, I think these two options should be enabled and tied to each other by default. 

Further, i would like to suggest a tiny edition of the description of this `slugify` option as follows.
`Standardize the slug of anchor links (pointing to heading title). Transform the slug into all lower case. Replace space with hyphen. Applicable only for anchor links in markdown link syntax`

I think that your plugin is the currently best companion for anyone who want to use Obsidian as a CMS (content management system) to publish with mkdocs-material theme.

**Reply by Mara-Li** [reply]
Thanks! I will change the description. Also thanks for the compliment :D
